pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -106,6 +107,40 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// RAIDVolume defines the desired configuration of volume in RAID
+type RAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.
+	// If unspecified or set be 0, the maximum capacity of disk will be used for logical disk.
+	SizeGb int `json:"sizeGb,omitempty"`
+
+	// RAID level for the logical disk. The following levels are supported: 0, 1, 1+0.
+	// +kubebuilder:validation:Enum=0,1,1+0
+	Level nodes.RAIDLevel `json:"level" required:"true"`
+
+	// Name of the volume. Should be unique within the Node. If not specified, volume name will be auto-generated.
+	Name string `json:"name,omitempty"`
+
+	// The valid values must be "hdd" or "ssd". If this is not specified, disk type will not be a criterion to find backing physical disks
+	// +kubebuilder:validation:Enum=hdd,ssd
+	DiskType nodes.DiskType `json:"diskType,omitempty"`
```

Comment:
If this field is optional, don't we need for it to be a pointer? Or does DiskType have a good zero value?

Reply:
Although Ironic will handle the zero value for DiskType, but I think it would be better if we make it to be a pointer
Thanks

Reply:
This still needs to be addressed. What is the zero value for nodes.DiskType? What default do we want for this field?

Reply:
I have no idea, but in OpenStack Docs, the a part of [example](https://docs.openstack.org/ironic/latest/admin/raid.html) haven't set disk_type, and i find some explanation.
>disk_type - hdd or ssd. If this is not specified, disk type will not be a criterion to find backing physical disks.

Reply:
Or we can check DiskType, if DiskType is empty return empty.

Reply:
It's [just a string](https://github.com/gophercloud/gophercloud/blob/master/openstack/baremetal/v1/nodes/requests.go#L524-L525) so the zero value is an empty string :man_shrugging:

Reply:
We probably do have to specifically allow an empty value in the kubebuilder validation of the previous line though I think?

Everywhere else in the API (HardwareDetails and root device hints) we have had a bool named Rotational instead of the strings "hdd" and "sdd". Let's do that here also and make this a bool pointer.